Assignment_1/palindrome.js

```diff
@@ -0,0 +1,46 @@
+class Node {
+    constructor(val) {
+        this.data = val;
+        this.next = null;
+    }
+}
+//
+const palindrome = (head) => {
+    if (!head || !head.next) {
+        return true; 
+    }//empty or one letter
+
+    let fast = head, slow = head;
+    const stack = [];
+
+    while (fast && fast.next) {
+        stack.push(slow.data);
+        fast = fast.next.next;
+        slow = slow.next; 
+    }//push first half of linked list onto stack
```

Comment:
Good use of the fast/slow pointer solution. If you wanted to, you could also do this with a for loop, taking advantage of JS's comma operator for the iteration part of the loop:
```suggestion
    for (; fast && fast.next; slow = slow.next, fast = fast.next.next) {
        stack.push(slow.data);
    }
```